http-service/src/main/java/net/runelite/http/service/feed/blog/BlogService.java

```diff
@@ -0,0 +1,132 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.http.service.feed.blog;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.http.api.RuneLiteAPI;
+import net.runelite.http.api.feed.FeedItem;
+import net.runelite.http.api.feed.FeedItemType;
+import net.runelite.http.service.util.exception.InternalServerErrorException;
+import net.runelite.http.service.util.exception.RemoteServerErrorException;
+import okhttp3.HttpUrl;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+@Service
+@Slf4j
+public class BlogService
+{
+	private static final HttpUrl RSS_URL = HttpUrl.parse("https://raw.githubusercontent.com/runelite/runelite.net/gh-pages/atom.xml");
+	private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US);
+
+	public List<FeedItem> getBlogPosts() throws IOException
+	{
+		Request request = new Request.Builder()
+			.url(RSS_URL)
+			.build();
+
+		try (Response response = RuneLiteAPI.CLIENT.newCall(request).execute())
+		{
+			if (!response.isSuccessful())
+			{
+				throw new RemoteServerErrorException("Error getting blog posts: " + response.message());
+			}
+
+			try
+			{
+				InputStream in = response.body().byteStream();
+				Document document = DocumentBuilderFactory.newInstance()
+						.newDocumentBuilder()
+						.parse(in);
+
+				Element documentElement = document.getDocumentElement();
+				NodeList documentItems = documentElement.getElementsByTagName("entry");
+
+				List<FeedItem> items = new ArrayList<>();
+
+				for (int i = 0; i < Math.min(documentItems.getLength(), 3); i++)
+				{
+					Node item = documentItems.item(i);
+					NodeList children = item.getChildNodes();
+
+					String title = null;
+					String summary = null;
+					String link = null;
+					long timestamp = -1;
+
+					for (int j = 0; j < children.getLength(); j++)
+					{
+						Node childItem = children.item(j);
+						String nodeName = childItem.getNodeName();
+
+						switch (nodeName)
+						{
+							case "title":
+								title = childItem.getTextContent();
+								break;
+							case "summary":
+								summary = childItem.getTextContent().replace("\n", "").trim();
+								break;
+							case "id":
+								link = childItem.getTextContent();
```

Comment:
I think correct way here to get the link would be getting it from the `<link href`, because in Atom specification id just needs to be unique, I put link there because it is unique but that might change in future, using `<link href` instead will be always correct.

Reply:
Also maybe renaming OSRSNewsService to RSS2ReaderService and BlogService to AtomReaderService would be better to make it more generic, because they should work for all RSS2 and Atom feeds in theory.

Reply:
I think it's not necessary to make generic RSS readers because it will only be used for this thing. I think it's just wasted time and space. If we will ever need to read more of these RSS/Atom feeds, we can maybe make generic readers.

Reply:
True, or use library that does that (which I can imagine there are tons of).